lib/libutee/tee_api_operations.c

```diff
@@ -763,39 +763,17 @@ static TEE_Result tee_buffer_update(
 
 	if (slen >= (buffer_size + buffer_left)) {
 		/* Buffer is empty, feed as much as possible from src */
-		if (TEE_ALIGNMENT_IS_OK(src, uint32_t)) {
```

Comment:
I agree that the SVC API doesn't specify any required alignment, but that can be fixed :-)
 
Has this been tested with `rijndael_ecb_encrypt()`?

Doing buffering in TEE Core will consume more memory there instead of using the memory in the TA. The TA will any way be prepared do some buffering since we only accept complete blocks.